server/neptune/workflows/internal/pr/revision/policy/handler.go

```diff
@@ -149,6 +155,34 @@ func (f *FailedPolicyHandler) fetchTeamMembers(ctx workflow.Context, repo gh.Rep
 	return listTeamMembersResponse.Members, err
 }
 
+// updateCheckStatuses checks to see if each root's workflow has any remaining failing policies.
+// If it doesn't then the workflow's check status reason is updated to be bypassed (i.e. passing)
+func (f *FailedPolicyHandler) updateCheckStatuses(ctx workflow.Context, roots map[string]revision.RootInfo, workflowResponses []terraform.Response, remainingFailedPolicies []activities.PolicySet) {
+	for _, workflowResponse := range workflowResponses {
+		// if this workflow's failing policies have been bypassed
+		if containsFailingPolicy(workflowResponse.ValidationResults, remainingFailedPolicies) {
+			continue
+		}
+		workflowState := workflowResponse.WorkflowState
+		workflowState.Result.Status = state.CompleteWorkflowStatus
+		workflowState.Result.Reason = state.BypassedFailedValidationReason
+		workflow.Go(ctx, func(c workflow.Context) {
+			f.Notifier.Notify(c, &workflowState, roots)
+		})
```

Comment:
I think you'll also want to wait on these after executing all of them otherwise there could be a race when a signal comes in right after.  Since there's no wait groups you might want to construct something simple that does essentially the same thing using temporal's await condition maybe.